### PR TITLE
fix: hot reloading + page refresh for section pagination

### PR DESCRIPTION
**Pull request by slavikbez**
## What/Why/How?

Fixes [#423](https://github.com/Redocly/openapi-cli/issues/423) [#424](https://github.com/Redocly/openapi-cli/issues/424) 

## Screenshots (optional)

https://user-images.githubusercontent.com/9461457/139879901-84f3e223-94c1-4023-b029-1ff9f3a973d6.mov

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts [hunk 1]
```diff
@@ -96,39 +100,45 @@ export default async function startPreviewServer(
...
+              `Something went wrong: ${e.code || e.message}...\n`,
+              request,
+              response,
+              {},
+              500,
+            );
+          }
+        }
+    } else {
+      const filePath = path.resolve(htmlTemplate ? path.dirname(htmlTemplate) : process.cwd(), `.${request.url}`);
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:132]: We lost functionality that serves any other files from the dir.

`filePath` variable is not used anywhere down.

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts [hunk 2]
```diff
@@ -96,39 +100,45 @@ export default async function startPreviewServer(
...
+            respondWithGzip(
+              `Something went wrong: ${e.code || e.message}...\n`,
+              request,
+              response,
+              {},
+              500,
+            );
+          }
+        }
+    } else {
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:131]: I would add the following check for serving index pages and would keep else code for serving other files unchanged.

```suggestion
    } else if (request.url.endsWith('/') ||  path.extname(request.url) === '') {
```

**Review by RomanHotsiy — CHANGES_REQUESTED** [review]
We lost functionality. Other files from `htmlTemplate` dir (or `cwd` if there is no `htmlTemplate`) should be served too (e.g. favicon.png, logo.png, etc).

**Comment by slavikbez** [comment]
> We lost functionality. Other files from `htmlTemplate` dir (or `cwd` if there is no `htmlTemplate`) should be served too (e.g. favicon.png, logo.png, etc).

Can there be files other than just static assets? For example if there is a `some-file.md` in the root, should it be accessible by `/some-file` in the browser, like we do it in portal?

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts (hunk 2 above)
Comment (slavikbez) [reply packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:131]: I think I got what you mean.

**Comment by RomanHotsiy** [comment]
> Can there be files other than just static assets? 

@slavikbez No, we do not support this as part of this project. Just static assets.

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts (hunk 1 above)
Comment (slavikbez) [reply packages/cli/src/commands/preview-docs/preview-server/preview-server.ts]: Fixed

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export default async function startPreviewServer(
     console.time(colorette.dim(`GET ${request.url}`));
     const { htmlTemplate } = getOptions() || {};
 
-    if (request.url === '/') {
+    if (request.url === '/' || request.url?.endsWith('/') || path.extname(request.url!) === '') {
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:67]:
```suggestion
    if (request.url?.endsWith('/') || path.extname(request.url!) === '') {
```

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts [hunk 4]
```diff
@@ -73,7 +73,7 @@ export default async function startPreviewServer(
           'Content-Type': 'text/html',
         },
       );
-    } else if (request.url === '/openapi.json') {
+    } else if (request.url?.endsWith('/openapi.json')) {
```
Comment (RomanHotsiy) [inline packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:76]: Is it always requested from the root?

**Review by RomanHotsiy — COMMENTED** [review]
Left a tiny suggestion and a question

File: packages/cli/src/commands/preview-docs/preview-server/preview-server.ts (hunk 4 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:76]: I think we should change `openapi.json` to `/openapi.json` here: https://github.com/Redocly/openapi-cli/blob/master/packages/cli/src/commands/preview-docs/preview-server/preview-server.ts#L49

Comment (slavikbez) [reply packages/cli/src/commands/preview-docs/preview-server/preview-server.ts:76]: yes, we can fix it there.

**Review by RomanHotsiy — APPROVED** [review]

**Pull request merged**